### PR TITLE
Added shape file online resource for AusCPR Zooplankton

### PR DESCRIPTION
**Pull request by bhasin85**
Related Portal [PR](https://github.com/aodn/aodn-portal/pull/2446)

### CODE REVIEW

File: iso19139.mcp-2.0/process/transform_testing_urls.xsl [hunk 1]
```diff
@@ -41,4 +41,34 @@
         </xsl:for-each>
     </xsl:template>
 
+    <xsl:param name="shape-file-online-resource">
+
+					<gmd:onLine>
```
Comment (jonescc) [inline iso19139.mcp-2.0/process/transform_testing_urls.xsl]: Perhaps indent as per other indenting - using tabs here elsewhere its spaces

File: iso19139.mcp-2.0/process/transform_testing_urls.xsl [hunk 2]
```diff
@@ -41,4 +41,34 @@
         </xsl:for-each>
     </xsl:template>
 
+    <xsl:param name="shape-file-online-resource">
```
Comment (jonescc) [inline iso19139.mcp-2.0/process/transform_testing_urls.xsl]: Is this really a parameter to the xsl?   Perhaps using a variable would be better.

File: iso19139.mcp-2.0/process/transform_testing_urls.xsl [hunk 3]
```diff
@@ -41,4 +41,34 @@
...
+					</gmd:onLine>
+    </xsl:param>
+
+    <xsl:variable name="expectedId" >9d6b28d2-2d6b-4726-b603-ab03e3d15d34</xsl:variable>
+
+    <xsl:template match="gmd:MD_DigitalTransferOptions">
+        <xsl:copy>
+            <xsl:copy-of select="@* | node()"/>
+            <xsl:if test="../../../../gmd:fileIdentifier/gco:CharacterString/text() = $expectedId">
+                <xsl:copy-of select="$shape-file-online-resource"/>
```
Comment (jonescc) [inline iso19139.mcp-2.0/process/transform_testing_urls.xsl]: Do we need to check if there is one already?  to cover the case where the change has been made in production but not yet applied to the systest transform

File: iso19139.mcp-2.0/process/transform_testing_urls.xsl (hunk 2 above)
Comment (bhasin85) [inline iso19139.mcp-2.0/process/transform_testing_urls.xsl]: Good Point. Updated to variable.

File: iso19139.mcp-2.0/process/transform_testing_urls.xsl (hunk 1 above)
Comment (bhasin85) [inline iso19139.mcp-2.0/process/transform_testing_urls.xsl]: Done :+1:

File: iso19139.mcp-2.0/process/transform_testing_urls.xsl (hunk 3 above)
Comment (bhasin85) [inline iso19139.mcp-2.0/process/transform_testing_urls.xsl]: Added the check.

**Comment by bhasin85** [comment]
@jonescc  Made the changes. :+1:

File: iso19139.mcp-2.0/process/transform_testing_urls.xsl [hunk 4]
```diff
@@ -26,7 +28,9 @@
     <!-- Convert point of truth links only to avoid accidental edits -->
 
     <xsl:template match="gmd:URL[../../gmd:protocol/*/text()='WWW:LINK-1.0-http--metadata-URL']">
-        <gmd:URL><xsl:value-of select="replace(.,'catalogue-imos', 'catalogue-systest')"/></gmd:URL>
+        <gmd:URL>
+            <xsl:value-of select="replace(.,'catalogue-imos', 'catalogue-systest')"/>
```
Comment (jonescc) [inline iso19139.mcp-2.0/process/transform_testing_urls.xsl]: This includes whitespace in the url which will cause problems

File: iso19139.mcp-2.0/process/transform_testing_urls.xsl [hunk 5]
```diff
@@ -36,9 +40,44 @@
 
         <xsl:for-each select="$urlSubstitutions/substitution">
             <xsl:if test="matches(string($url), string(@match))">
-                <gmd:URL><xsl:value-of select="replace($url, string(@match), string(@replaceWith))"/></gmd:URL>
+                <gmd:URL>
+                    <xsl:value-of select="replace($url, string(@match), string(@replaceWith))"/>
```
Comment (jonescc) [inline iso19139.mcp-2.0/process/transform_testing_urls.xsl]: And here

**Comment by bhasin85** [comment]
@jonescc  Closing PR as the changes are already made in production.

**Pull request closed without merging**